dev/core/src/com/google/gwt/core/linker/SymbolMapsLinker.java

```diff
@@ -335,6 +358,70 @@ public Object merge(String extKey, Object oldVal, Object newVal) {
     return artifacts;
   }
 
+  private static void embedSourcesInSourceMaps(TreeLogger logger, StandardLinkerContext context,
+                                               ArtifactSet artifacts,
+                                               SourceMapGeneratorV3 sourceMapGenerator,
+                                               int totalPrefixLines, String sourceMapString,
+                                               String partialPath)
+      throws SourceMapParseException {
+    sourceMapGenerator.setStartingPosition(totalPrefixLines, 0);
+    SourceMapConsumerV3 section = new SourceMapConsumerV3();
+    section.parse(sourceMapString);
+    section.visitMappings(sourceMapGenerator::addMapping);
+
+    Iterator extensions = section.getExtensions().entrySet().iterator();
+
+    while (extensions.hasNext()) {
+      Entry<String, Object> entry = (Entry) extensions.next();
+      String extensionKey = (String) entry.getKey();
+      sourceMapGenerator.addExtension(extensionKey, entry.getValue());
+    }
+
+    ResourceLoader resourceLoader = ResourceLoaders.fromContextClassLoader();
+
+    Map<String, EmittedArtifact> generatedSources = new java.util.HashMap<>();
```

Comment:
import HashMap - or probably more consistent with the rest of the codebase, use the factory method in Maps?